### ISSUE TITLE
Accessing public files raises error

### ISSUE THREAD

**Issue by Conchylicultor** [christopher-hesse/blobfile#195]
I'm trying to access a public `gs://` file. However blobfile fails with credential missing error.

Reproduction instruction:

```python
>>> blobfile.exists('gs://tfds-data/datasets/mnist/3.0.1/dataset_info.json')
blobfile._common.Error: credentials not found, please login with 'gcloud auth application-default login'
or else set the 'GOOGLE_APPLICATION_CREDENTIALS' environment variable to the
path of a JSON format service account key
```
By comparison:
```python
>>> tf.io.gfile.exists('gs://tfds-data/datasets/mnist/3.0.1/dataset_info.json')
True
```

The file is public so shouldn't need any permissions for read access.

I'm using `blobfile.__version__ == '1.3.0'`, tested on MacOS

**Comment by christopher-hesse** [comment]
Try the newest master branch and reopen this if it doesn't work for you.

**Comment by Conchylicultor** [comment]
Thank you for the quick fix :)
I confirm this work when installed from HEAD

Considering this is quite a major bug, I hope the 1.3.1 can be released soon!

**Comment by christopher-hesse** [comment]
I appreciate the feature request but I should update the description
because I realized it’s confusing. This isn’t really a clone of tf.gfile,
and I don’t think there are any tests that verify the behavior is the same.

Reading public files is sort of an edge case because most of those can be
read via http and you don’t need blobfile for that.

In fact this new mode means someone will likely get a confusing error when
they attempt to read a private file but didn’t set their credentials
correctly, so it could be that adding support for anonymous access is a bad
move here.

However I did add support for azure public containers so it would be more
consistent to also support google public files.
